Implement TaskListScreen with LazyColumn

baraa0abd opened the pull request:
- Implemented `PriorityChip` composable to display task priority with an icon.
- Implemented `TaskItem` composable to display individual task details, including an icon, title, description, and priority chip.
- Created `TaskListScreen` composable that uses `LazyColumn` to display a list of `TaskItemData`.
- Added sample data for previewing the `TaskListScreen`.




![image](https://github.com/user-attachments/assets/3b464c72-d5c6-4d77-a8d9-a9f21ae6c30a)

ABDULLAHHG committed:
add nunito font styles to resource

ABDULLAHHG committed:
add a data class for tudee text style

Arzo-zi committed:
add tudeeColor to handle light and dark theme

Arzo-zi committed:
refactor theme management for light and dark

ABDULLAHHG committed:
implement tudee text style font

ABDULLAHHG committed:
Merge branch 'feature/color-design-lightmode' into feature/design-system-font-declear

ABDULLAHHG committed:
Refactor textStyle package location and integrate typography into theme

Hasa9nAli committed:
call content of application inside TudeeTheme tree

Hasa9nAli committed:
add all drawable icons of project

baraa0abd committed:
Add deployment target selector configuration and robot images

This commit introduces the following:
- Adds `.idea/deploymentTargetSelector.xml` file to configure deployment target selection for the app run configuration.
- Adds various density versions (mdpi, hdpi, xhdpi, xxhdpi, xxxhdpi) of three different robot images:
    - `749d09de62af3aa8ec816c102a9dce71959de02a.png` (happy robot)
    - `Frame 2095586230.png` (frame)
    - `Icon.png` (icon)
- Organizes these images into `robotimages`, `robotimage2`, and `robotimage3` subdirectories within `app/src/main/res/drawable/`.
- Also adds these images directly into density-specific drawable folders (e.g., `app/src/main/res/drawable-hdpi/`) with descriptive names like `img_happy_robot.png`, `img_frame_hdpi.png`, and `img_icon_hdpi.png`.

Hasa9nAli committed:
add SnakeBar composable for displaying messages with status indicators

Hasa9nAli committed:
Merge remote-tracking branch 'origin/feature/design-system' into feature/design-system

Hasa9nAli committed:
Add DayChip composable for displaying selectable day options

Hasa9nAli committed:
Add Priority composable for displaying task priority indicators

ABDULLAHHG committed:
ignore the whole .idea

ABDULLAHHG committed:
change compile sdk and target sdk to 35

baraa0abd committed:
Implement TaskListScreen with LazyColumn

This commit introduces the `TaskListScreen` composable, which displays a list of tasks using `LazyColumn` for efficient rendering.

Key changes:
- Created `TaskItemData` data class to hold task information (icon, title, description, priority).
- Implemented `PriorityChip` composable to display task priority with an icon.
- Implemented `TaskItem` composable to display individual task details, including an icon, title, description, and priority chip.
- Created `TaskListScreen` composable that uses `LazyColumn` to display a list of `TaskItemData`.
- Added sample data for previewing the `TaskListScreen`.

baraa0abd committed:
Add new drawable icons and update existing ones

This commit introduces several new drawable resources:
- ic_pur_book.xml
- ic_plant.xml
- ic_baseball.xml
- ic_programming.xml
- ic_bag.xml
- ic_shopping.xml
- user_multiple.xml
- ic_gym.xml
- ic_quran_blue.png (binary file)
- img.png (binary file)
- img_1.png (binary file)
- img_2.png (binary file)
- img_3.png (binary file)
- img_4.png (binary file)
- img_5.png (binary file)

It also updates the `ic_chef.xml` drawable with minor path data and viewport size changes.

The `Category.kt` composable has been updated to use these new and updated icons in the sample task data.

baraa0abd committed:
Add developer and quran icons and update TaskItem composable

- Added `ic_developer.xml` and `ic_quran2.xml` drawable resources.
- Introduced a new composable `TaskInfo` to display task title and description.
- Refactored `TaskItem` composable to utilize the new `TaskInfo` composable for better structure.
- Updated icon references in `TaskItemData` within `CategoryScreen` to use the new and existing icons.
- Changed the background color of the `LazyColumn` in `CategoryScreen` to white.

baraa0abd committed:
Remove obsolete drawable resources

This commit removes several drawable resources that are no longer needed. The deleted files include:

- `749d09de62af3aa8ec816c102a9dce71959de02a.png` from various density folders (`drawable-hdpi`, `drawable-mdpi`, `drawable-xhdpi`, `drawable-xxhdpi`, `drawable-xxxhdpi`) within `robotimages/android/`.
- `Frame 2095586230.png` from various density folders (`drawable-hdpi`, `drawable-mdpi`, `drawable-xhdpi`, `drawable-xxhdpi`, `drawable-xxxhdpi`) within `robotimage2/android/`.
- `frame_hdpi_xxhdpi.png` from `drawable-xxhdpi`.
- `ic_quran_blue.png` from `drawable`.
- `Icon.png` from various density folders (`drawable-hdpi`, `drawable-mdpi`, `drawable-xhdpi`, `drawable-xxhdpi`, `drawable-xxxhdpi`) within `robotimage3/android/`.
- `img.png`, `img_1.png`, `img_2.png`, `img_3.png`, `img_4.png`, `img_5.png` from `drawable`.
- `img_frame_hdpi.png` from `drawable-hdpi`.
- `img_frame_mdpi.png` from `drawable-mdpi`.
- `img_frame_xhdpi.png` from `drawable-xhdpi`.
- `img_frame_xxxhdpi.png` from `drawable-xxxhdpi`.
- `img_happy_robot.png` from `drawable-hdpi`.
- `img_happy_robot2.png` from `drawable-mdpi`.
- `img_happy_robot_xh.png` from `drawable-xhdpi`.
- `img_happy_robot_xxhdpi.png` from `drawable-xxhdpi`.
- `img_happy_robot_xxxhdpi.png` from `drawable-xxxhdpi`.
- `img_icon_hdpi.png` from `drawable-hdpi`.
- `img_icon_mdpi.png` from `drawable-mdpi`.
- `img_icon_xhdpi.png` from `drawable-xhdpi`.
- `img_icon_xxhdpi.png` from `drawable-xxhdpi`.
- `img_icon_xxxhdpi.png` from `drawable-xxxhdpi`.

baraa0abd committed:
Refactor: Remove redundant comments from Category composable

This commit removes unnecessary comments from the `Category.kt` file. The comments were explaining standard Modifier attributes or self-explanatory code, and their removal improves code readability.

baraa0abd committed:
Refactor: Remove redundant comments from Category composable

This commit removes unnecessary comments from the `Category.kt` file. The comments were explaining standard Modifier attributes or self-explanatory code, and their removal improves code readability.

ABDULLAHHG committed:
readd ignore the whole .idea directory

baraa0abd committed:
Implement TaskListScreen with LazyColumn

This commit introduces the `TaskListScreen` composable, which displays a list of tasks using `LazyColumn` for efficient rendering.

Key changes:
- Created `TaskItemData` data class to hold task information (icon, title, description, priority).
- Implemented `PriorityChip` composable to display task priority with an icon.
- Implemented `TaskItem` composable to display individual task details, including an icon, title, description, and priority chip.
- Created `TaskListScreen` composable that uses `LazyColumn` to display a list of `TaskItemData`.
- Added sample data for previewing the `TaskListScreen`.

baraa0abd committed:
Refactor TaskItem and TaskListScreen composables

This commit refactors the `TaskItem` and `TaskListScreen` composables.

- In `TaskItem`, comments for fixed width, height, background color, shape, border, and padding have been removed as the code is self-explanatory.
- In `TaskListScreen`, the extensive sample data list has been significantly shortened to a single item for brevity. The comment indicating that data would come from a ViewModel in a real app has also been removed.

baraa0abd committed:
Refactor TaskItem and introduce TaskInfo composable

This commit refactors the `TaskItem` composable by extracting the title and description display logic into a new composable called `TaskInfo`.

Key changes:
- Created `TaskInfo` composable to display task title and description with specific styling.
- Updated `TaskItem` to utilize the new `TaskInfo` composable.
- Changed the icon used in the sample task data within `TaskListScreen` from `ic_birthday_cake` to `ic_pur_book`.
- Changed the background color of the `LazyColumn` in `TaskListScreen` to white.

baraa0abd committed:
Refactor PriorityChip and adjust Category layout

- Changed `Icon` to `Image` in `PriorityChip` to allow use of drawable with inherent colors.
- Removed explicit tint from the priority icon.
- Adjusted vertical offset in `Category` composable for better alignment.
- Minor code formatting changes in `Category.kt`.

baraa0abd committed:
add content to pics

baraa0abd committed:
Refactor TaskItem into SubTaskItem composable

This commit extracts the UI logic for displaying a single task item from the `TaskItem` composable into a new reusable composable named `SubTaskItem`.

The `TaskItem` composable now utilizes `SubTaskItem` to render individual task details. This improves code organization and reusability.

Malak187 commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
Please note that in the original design (as shown in the image below), the book image is vertically aligned with the text.
However, in your current implementation, the image appears to start before the text, causing a misalignment.

![image](https://github.com/user-attachments/assets/4767e05f-1408-49c6-8a2b-181a689b276b)

Malak187 reviewed (commented):
Please note that in the original design (as shown in the image below), the book image is vertically aligned with the text.
However, in your current implementation, the image appears to start before the text, causing a misalignment.


![image](https://github.com/user-attachments/assets/117b7b4e-691f-46a2-bbeb-430c4d1d8616)

baraa0abd committed:
Refactor TaskItem and SubTaskItem composables

This commit refactors the `TaskItem` and `SubTaskItem` composables.

- `SubTaskItem` has been renamed to `TaskImage`.
- The `Image` component within `PriorityChip` has been replaced with an `Icon` component, and its tint is set to `Color.White`.
- Layout adjustments have been made to `TaskImage` and `TaskItem` using `offset` and `padding` modifiers for better visual alignment.

Farah-Dev4 commented on app/src/main/java/com/baghdad/tudee/ui/composable/SnakeBar.kt:
```diff
@@ -0,0 +1,85 @@
+import com.baghdad.tudee.designSystem.theme.TudeeTheme
+
+@Composable
+fun SnakeBar(
```
I have a confused about the function name "SnakeBar" because I search about it and its appear "SnackBar" also in figma its wrote like "SnackBar"

baraa0abd replied:
Actually, it's not my task 😅

baraa0abd replied:
![image](https://github.com/user-attachments/assets/7ce5ffde-e6ef-4e79-acd9-d3892f643c02)
 but I think this one

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.White)
```
use color theme please

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+            .width(320.dp)
+            .height(111.dp)
+            .background(
+                color = Color(0xFFFFFFFF),
```
use color theme please

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+            modifier = Modifier
+                .offset(x = 8.dp, y = 0.dp)
+                .fillMaxWidth(),
+            color = Color(0xFF1F1F1F).copy(alpha = 0.6f),
```
use color theme please , don't use change color (apha)here please  (follow the design system)

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+                .width(296.dp)
+                .height(19.dp)
+                .fillMaxWidth(),
+            color = Color(0x991F1F1F),
```
use color theme please

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+        )
+        Text(
+            text = text,
+            color = Color.White,
```
use color theme please

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+    Row(
+        modifier = modifier
+            .background(
+                color = Color(0xFFF2C849),
```
use color theme please

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+import com.baghdad.tudee.R
+
+data class TaskItemData(
+    @DrawableRes val iconRes: Int,
```
use `Painter` data type instead  `@DrawableRes `

Hasa9nAli commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+            style = TextStyle(
+                lineHeight = 19.sp,
+                letterSpacing = 0.sp
+            )
```
follow the design system please

MuhammedEdrees replied:
Same thing use the theme typography and the predefined style.

Hasa9nAli reviewed (changes requested):
Great Work Baraa, 
please show the comment

MuhammedEdrees commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+            text = text,
+            color = Color.White,
+            fontSize = 14.sp,
+            fontWeight = FontWeight.Medium
```
If you click on each text component in the figma design, you'll see it has the name of the style applied. This style is already added in the design system so you can use it directly instead of specifying text style in each text separately for example the picture below is from the priority chip text, the style is LabelSmall. Same thing for the other text components.
![image](https://github.com/user-attachments/assets/b30af3a4-6c95-4645-9e2a-9d1ce17c9b3c)

MuhammedEdrees commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+            fontSize = 12.sp,
+            textAlign = TextAlign.Start,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis
```
Same issue.

baraa0abd committed:
Refactor colors and fontStyle by following the design sys

MuhammedEdrees commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,221 @@
+}
+
+@Composable
+fun TaskListScreen() {
```
Naming this `TaskListScreen` could a bit misleading as it doesn't represent an actual screen. Naming it `TasksList` should be sufficient. Also, I'm not sure what is the use for this component is it for testing purposes? because it doesn't take any parameters and therefore cannot be used as a component.

MuhammedEdrees replied:
Also small note, the file name is `Category`. is it related to the components inside it?

baraa0abd replied:
yes It's or testing purpose , also to implement list scrolling , the file Category refers to taskCategory

MuhammedEdrees replied:
I think you should remove it then before merging to `develop` if it's for testing purposes.

MuhammedEdrees reviewed (changes requested):
Overall, Great work baraa. 👏 Just some small adjustments and you're there!

baraa0abd committed:
Refactor TaskItem and related composables for improved structure and styling

This commit introduces several changes to the `TaskItem` composable and its related components:

- **Refactored `TaskItem`:**
    - The internal structure of `TaskItem` has been simplified.
    - `TaskImage`, `TaskInfo`, and `PriorityChip` are now direct children of the main `Row` within `TaskItem`.
    - Removed nested `Row` and `Box` layouts.
- **Updated `PriorityChip`:**
    - Uses `Theme.color.status.yellowAccent` for background and `Theme.color.surfaceColor.surfaceHigh` for icon tint.
    - Text now uses `tudeeTextStyle.label.small` and `Theme.color.textColor.onPrimary`.
- **Updated `TaskInfo`:**
    - Title text now uses `Theme.color.textColor.body` and `tudeeTextStyle.body.medium`.
    - Description text now uses `Theme.color.textColor.title` and `tudeeTextStyle.label.small`.
    - Removed explicit `offset` modifiers.
- **Updated `TaskListScreen` (renamed to `TaskList`):**
    - The main composable function `TaskListScreen` has been renamed to `TaskList`.
    - The sample data in `TaskList` now uses `Theme.color.status.pinkAccent` for the first item's icon background and `ic_bag` drawable.
    - Background color of the `LazyColumn` now uses `Theme.color.surfaceColor.surfaceHigh`.
- **General Styling:**
    - Replaced direct `lightThemeColor` references with `Theme.color` for better theming support.
    - Standardized text styling using `tudeeTextStyle`.
- **Removed Unused Imports:**
    - `androidx.annotation.DrawableRes`
    - Specific text style imports like `TextStyle`, `FontWeight`, `TextAlign`.

ABDULLAHHG commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,226 @@
+    val priority: String
+)
+
+@Composable
```
you did not used iconRes or iconBackgoundColor
you forget to use iconRes to make the image change depend on the input not spacify it in the code 
you make it static in line 159

ABDULLAHHG commented:
you should implement if there is date in the task

ABDULLAHHG commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,198 @@
+            contentDescription = item.title,
+            modifier = Modifier.size(56.dp)
+        )
+
```
here the image is static

baraa0abd committed:
Refactor TaskItem and TaskImage composables for priority display

This commit refactors the `TaskItem` and `TaskImage` composables to utilize the new `Priority` composable for displaying task priority.

Key changes:
- Removed the `PriorityChip` composable.
- Updated `TaskItem` to pass a `PriorityTask` enum value (HIGH, MEDIUM, LOW) to the `Priority` composable based on the `item.priority` string.
- Modified `TaskImage` to accept an `imagePainter` parameter and also use the `Priority` composable.
- In the sample data within `TaskList`, the icon for the "Organize Birthday Party" task has been changed from `ic_bag` to `ic_alert`.

baraa0abd committed:
Remove unused TaskList and TaskListScreenPreview composables

The `TaskList` and `TaskListScreenPreview` composables, along with their associated task data, have been removed from `Category.kt` as they are no longer in use.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feature/task-category

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/baghdad/tudee/designSystem/color/TudeeColor.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/textStyle/TudeeTextStyle.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/textStyle/defaultTextStyle.kt
#	app/src/main/java/com/baghdad/tudee/designSystem/theme/TudeeTheme.kt
#	app/src/main/java/com/baghdad/tudee/ui/composable/DayChip.kt
#	app/src/main/java/com/baghdad/tudee/ui/composable/SnakeBar.kt

baraa0abd committed:
refactor the category

ahmednasserzaza commented on app/src/main/java/com/baghdad/tudee/ui/composable/Category.kt:
```diff
@@ -0,0 +1,71 @@
+        modifier = modifier
+            .fillMaxWidth()
+            .background(Theme.color.surfaceColor.surfaceHigh, shape = RoundedCornerShape(16.dp))
+            .padding(horizontal = 12.dp, vertical = 12.dp)
```
I see padding of horizontal = vertical, just pass 12.dp directly

ahmednasserzaza reviewed (approved):
I added a new comment, please check it

baraa0abd committed:
Rename file from Category to CategoryTaskCard and implement review

MuhammedEdrees reviewed (approved):
Well done 👏

Hasa9nAli committed:
Merge branch 'develop' into feature/task-category
